ivy/functional/backends/torch/elementwise.py

```diff
@@ -60,7 +60,8 @@ def imag(
     out: Optional[torch.Tensor] = None,
 ) -> torch.Tensor:
     if val.dtype not in (torch.complex64, torch.complex128):
-        return torch.zeros_like(val, dtype=val.dtype)
+        ret = torch.imag(val.to(torch.complex64))
```

Comment:
I don't think we should cast both `complex64` and `complex128` to `complex64`, we might clip large values

Reply:
yeah make sense

Reply:
here we are casting dtypes other than `complex` to `complex64`,
I'll refactor it to make it more readble.

Reply:
Yeah that's an interesting point tbh, do we want to casting all other dtypes to `complex64` or just mark complex dtypes as supported for this backend if `torch.imag` doesn't support any other dtypes than `complex` ones as mentioned in their [docs](https://pytorch.org/docs/stable/generated/torch.imag.html). What do you think? Thanks @Madjid-CH 😄

Reply:
if we don't want any additional memory overhead like we did with the other functions we can mark the only supported types. And I don't see any reason why this function should be an exception. So let's just mark `complex` as supported.
@MahmoudAshraf97 what do you think ?